src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -740,4 +740,159 @@ var fluid = fluid || fluid_2_0_0;
+    /**
+     *
+     * Convert a string to a Boolean, for example, when working with HTML form element values.
+     *
+     * The following are all false: undefined, null, "", "0", "false", false, 0
+     *
+     * Everything else is true.
+     *
+     * @param value {String} The value to be interpreted.
+     * @returns {Boolean} The interpreted value.
+     */
+    fluid.transforms.stringToBoolean = function (value) {
+        if (value) {
+            return !Boolean(value === "" || value === "0" || value === "false");
+        }
+        else {
+            return false;
+        }
+    };
+
+    fluid.defaults("fluid.transforms.stringToBoolean", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Convert any value into a stringified boolean, i. e. either "true" or "false".  Anything that evaluates to
+     * true (1, true, "non empty string", {}, et. cetera) returns "true".  Anything else (0, false, null, et. cetera)
+     * returns "false".
+     *
+     * @param value - The value to be converted to a stringified Boolean.
+     * @returns {string} - A stringified boolean representation of the value.
+     */
+    fluid.transforms.booleanToString = function (value) {
+        return value ? "true" : "false";
+    };
+
+    fluid.defaults("fluid.transforms.booleanToString", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Transform stringified JSON to an object using `JSON.parse`.  Returns `undefined` if the JSON string is invalid.
+     *
+     * @param value {String} - The stringified JSON to be converted to an object.
+     */
+    fluid.transforms.JSONstringToObject = function (value) {
+        try {
+            return JSON.parse(value);
+        }
+        catch (e) {
+            return undefined;
+        }
+    };
+
+    fluid.defaults("fluid.transforms.JSONstringToObject", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Transform an object to a string using `JSON.stringify`.  You can pass the `space` option to be used
+     * as part of your transform, as in:
+     *
+     * ```
+     * "": {
+     *   transform: {
+     *     funcName: "fluid.transforms.objectToJSONString",
+     *     inputPath: "",
+     *     space: 2
+     *   }
+     * }
+     * ```
+     *
+     * The default value for `space` is 0, which disabled spacing and line breaks.
+     *
+     * @param value {Object} - An object to be converted to stringified JSON.
+     *
+     */
+    fluid.transforms.objectToJSONString = function (value, transformSpec) {
+        var space = transformSpec.space || 0;
+        return JSON.stringify(value, null, space);
+    };
+
+    fluid.defaults("fluid.transforms.objectToJSONString", {
+        gradeNames: ["fluid.standardTransformFunction"]
+    });
+
+    /**
+     *
+     * Transform a string to a date using the Date constructor.  Accepts (among other things) the date and dateTime values
```

Comment:
Cheers - I think this is very nearly right. I agree that there needs to be some extension of the model idiom that assists more in dealing with "the outside world" - Clayton's wording for this has been "materialisation" which we've fretted about a bit since it implies that things inside the model are immaterial and things outside it aren't - which we don't quite agree with. I think we should just be a bit clearer about the status quo - it's not that the framework doesn't guarantee that Dates are not handled correctly now - in fact it does - it's just that they will probably not be handled correctly in future without some further formality. The docs you reference bear this out - "a tree of objects has the status of being a model if it can successfully be copied by the framework utility function fluid.copy()" - which *can* include Date objects if the user is happy to treat these as immutable. But we need to suggest that i) this is just the kind of subtlety that we need to indicate that the user will need to take care of for themselves, and ii) that this practice should be considered in some sense "already deprecated" in that it might cause their code to break in the face of future framework changes.
I guess what you're getting at with your "outside the model" is that we plan to licence this use via the "fluid.model.transformWithRules" API for static documents - which is certainly one possible usage. But all the same provisos apply there too - if the user puts Dates into their static documents, they will need to be prepared to tangle the fact that they will end up aliased to any equivalents in the output, which is just the same issue with models. So I'm not sure that the "outside the model" hedge really makes it clear to the users what they are letting themselves in for.